src/commands/publish/upload_form/wasm_module.rs

```diff
@@ -1,19 +1,21 @@
+use std::path::PathBuf;
+
 use failure::format_err;
 
 use super::binding::Binding;
 use super::filename_from_path;
 
 #[derive(Debug)]
 pub struct WasmModule {
-    path: String,
+    path: PathBuf,
     filename: String,
     binding: String,
 }
 
 impl WasmModule {
-    pub fn new(path: String, binding: String) -> Result<Self, failure::Error> {
+    pub fn new(path: PathBuf, binding: String) -> Result<Self, failure::Error> {
         let filename = filename_from_path(&path)
-            .ok_or_else(|| format_err!("filename should not be empty: {}", path))?;
+            .ok_or_else(|| format_err!("filename should not be empty: {:?}", path))?;
```

Comment:
iirc this sucks to read on windows, suggestions welcome

Reply:
^ maybe my `{}` and `path.display()` suggestion above will address this?

Reply:
it should ^ !!